File: testprojects/scala_main/pipelines/sony_livy_clone/code/src/main/scala/io/prophecy/pipelines/livy_scala/udfs/UDFs.scala
package io.prophecy.pipelines.livy_scala.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {
  var a = 10

  def registerUDFs(spark: SparkSession) =
    spark.udf.register("udf1", udf1)

  def udf1 = udf((value: Int) => value * a)
}

File: testprojects/python_main/pipelines/LIVY_PYTHON/code/livy_python/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)
a = 10

def registerUDFs(spark: SparkSession):
    spark.udf.register("udf1", udf1)

@udf(returnType = IntegerType())
def udf1(value):
    return value * a if value != None else a * a
